src/grt/src/fastroute/src/RipUp.cpp

```diff
@@ -443,6 +443,53 @@ bool FastRouteCore::newRipup3DType3(const int netID, const int edgeID)
   return true;
 }
 
+void FastRouteCore::releaseResourcesNet(const int netID)
+{
+  const int edgeCost = nets_[netID]->edgeCost;
+  const TreeEdge* treeedges = sttrees_[netID].edges;
+  const int deg = sttrees_[netID].deg;
+
+  for (int edgeID = 0; edgeID < 2 * deg - 3; edgeID++) {
+
+    const TreeEdge* treeedge = &(treeedges[edgeID]);
+    const std::vector<short>& gridsX = treeedge->route.gridsX;
+    const std::vector<short>& gridsY = treeedge->route.gridsY;
+    const std::vector<short>& gridsL = treeedge->route.gridsL;
+    const int routeLen = treeedge->route.routelen;
+
+    for (int i = 0; i < routeLen; i++) {
+
+      if(gridsL[i] != gridsL[i + 1])
+        continue;
+
+      else if (gridsX[i] == gridsX[i + 1]) {  // a vertical edge
+
+        const int ymin = std::min(gridsY[i], gridsY[i + 1]);
+        if (v_edges_[ymin][gridsX[i]].usage >= edgeCost)
```

Comment:
Get v_edges_[ymin][gridsX[i]] and store it in a reference variable instead of repeatedly looking it up.  Same in the horiz case.  I think you could collapse the H/V cases by doing so